iocore/net/SSLNetVConnection.cc

```diff
@@ -546,7 +546,7 @@ SSLNetVConnection::net_read_io(NetHandler *nh, EThread *lthread)
   // If the key renegotiation failed it's over, just signal the error and finish.
   if (sslClientRenegotiationAbort == true) {
     this->read.triggered = 0;
-    readSignalError(nh, static_cast<int>(r));
+    readSignalError(nh, (errno) ? errno : -ENET_SSL_FAILED);
```

Comment:
Which function's error is this checking?

Reply:
This is based on the regeneration failure set in ssl_callback_info.  Probably best as just ENET_SSL_FAILED rather than trying to pick up the random errno along the way.